File: backend/src/main/java/com/propcoza/legends/tools/rental_transaction_manager/dto/AgentReturnDto.java
package com.propcoza.legends.tools.rental_transaction_manager.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AgentReturnDto {
    private String fullName;

    private String email;

    private String bankName;

    private String accountNumber;

    private String branchCode;

    private Boolean isActive;

    private String createdBy;

    private LocalDateTime createdAt;

    private String lastModifiedBy;

    private LocalDateTime lastModifiedAt;


    // Returns the count of rentals rather than the full list for performance
    private Integer totalRentals;
}

File: frontend/src/features/agents/agentReturnDto.ts
export interface AgentReturnDto {
  id: string
  fullName: string;
  email: string;
  bankName: string;
  accountNumber: string;
  branchCode: string;
  isActive: boolean;
  createdBy: string;
  createdAt: string; // ISO 8601 strings are standard for LocalDateTime
  lastModifiedBy: string;
  lastModifiedAt: string;
  totalRentals: number;
}